ProcessLib/BoundaryCondition/BoundaryConditionCollection.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_BOUNDARYCONDITIONCOLLECTION_H
+#define PROCESSLIB_BOUNDARYCONDITIONCOLLECTION_H
+
+#include "DirichletBoundaryCondition.h"
+#include "ProcessLib/ProcessVariable.h"
+
+namespace ProcessLib
+{
+
+class BoundaryConditionCollection : public BoundaryCondition
```

Comment:
Needs discussion: I'm not very happy with the statement "A boundary conditions collection is a boundary condition."
There must be an anchor for the recursive definition...